.github/workflows/ci.yml

```diff
@@ -0,0 +1,54 @@
+name: CI
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+  # Run daily at 1:23 UTC
+  schedule:
+  - cron:  '23 1 * * *'
+  workflow_dispatch:
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  test:
+    name: "test upload via action"
+    runs-on: ubuntu-latest
+    if: github.repository == 'scientific-python/upload-nightly-action'
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.x'
+
+      - name: Install python-build and twine
+        run: |
+          python -m pip install --upgrade pip
+          python -m pip install build twine
+          python -m pip list
+
+      - name: Build a wheel and a sdist
+        run: |
+          PYTHONWARNINGS=error,default::DeprecationWarning python -m build --outdir ./dist tests/test_package
+
+      - name: Verify the distribution
+        run: twine check --strict dist/*
+
+      - name: List contents of sdist
+        run: python -m tarfile --list dist/test-package-*.tar.gz
+
+      - name: List contents of wheel
+        run: python -m zipfile --list dist/test_package-*.whl
+
+      - name: Test upload
+        uses: ./
+        with:
+          artifacts_path: dist
+          anaconda_nightly_upload_token: ${{ secrets.UPLOAD_TOKEN }}
```

Comment:
So where would this test package end up being uploaded? Is there any cleanup planned once the upload is successful?

Reply:
> So where would this test package end up being uploaded?

c.f. https://github.com/scientific-python/upload-nightly-action/pull/27#issuecomment-1645021980

> Is there any cleanup planned once the upload is successful?

No, but does there need to be?

Reply:
Yes. I would definitely like to avoid polluting the user deployed space with testing artifacts.

Reply:
I'm not suggesting another location, but to do a cleanup. Easy and simple and the pollution will be only present when there is a problem.